ccdproc/image_collection.py

```diff
@@ -51,17 +51,30 @@ class ImageFileCollection(object):
         list.
         Default is ``None``.
 
+    filenames: str, list of str, or None, optional
+        List of the names of FITS files which will be added to the collection.
+        The filenames are assumed to be in ``location``.
+        Default is ``None``.
+
     Raises
     ------
     ValueError
         Raised if keywords are set to a combination of '*' and any other
         value.
     """
-    def __init__(self, location=None, keywords=None, info_file=None):
+    def __init__(self, location=None, keywords=None, info_file=None,
+                 filenames=None):
         self._location = location
+        self._filenames = filenames
         self._files = []
         if location:
-            self._files = self._fits_files_in_directory()
+            if self._filenames:
+                if type(self._filenames) is str:
```

Comment:
I think there is some convention that one compares types using `isinstance`: `isinstance(self._filenames, six.string_types)` (not sure if it's really called that 😄 )